### PR TITLE
Add support of the 2fa

### PR DESCRIPTION
**Pull request by iskandersamerkhanov**
👓 @pwildani @schwers @tomsoir

Along with these changes are the changes in reddit-mobile repo https://github.com/reddit/reddit-mobile/pull/1127

JIRA Tickets:
https://reddit.atlassian.net/browse/AE-1021
https://reddit.atlassian.net/browse/AE-1024

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -117,10 +117,18 @@ var login = function login(apiOptions, username, pass, orderedHeaders, clientUse
...
+      if (details === 'TWO_FA_REQUIRED'){
+        resolve(details);
+      }
+
       var errors = parseErrors(res);
       if (errors.length) {
         if (Array.isArray(errors[0]) && !!errors[0][0]) {
           var firstError = errors[0][0];
+          if (firstError === 'WRONG_PASSWORD' && errors[0].length > 2 && errors[0][2] === 'otp') {
+            return reject('WRONG_OTP');
```
Comment (pwildani) [inline index.js]: We could also add a WRONG_OTP error to the desktop login form instead of adding a weird special case here.

Comment (iskandersamerkhanov) [reply index.js]: will do it

**Comment by pwildani** [comment]
The API is now sending a WRONG_OTP and a reason (thanks!) is this PR still needed?

**Comment by iskandersamerkhanov** [comment]
Removed code that adds special case of WRONG_PASSWORD and rejects WRONG_OTP since we added WRONG_OTP error to desktop login

**Review by schwers-zz — APPROVED** [review]

**Pull request merged**